master/logging.lua

```diff
@@ -0,0 +1,17 @@
+local util = require "master.util"
+
+local state = util.mesos_get_state()
+if state == nil then
+    ngx.status = ngx.HTTP_SERVICE_UNAVAILABLE
+    ngx.say("503 Service Unavailable: invalid Mesos state.")
+    return ngx.exit(ngx.HTTP_SERVICE_UNAVAILABLE)
+end
+for _, agent in ipairs(state["slaves"]) do
+    if agent["id"] == ngx.var.agentid then
```

Comment:
> we should probably make it generic so auth.lua and logging.lua can reuse the same bits
Do you mean `agent.lua` instead of `auth.lua` ?

> I am just not quite sure what would be the best way of doing it.
We can make this lua code set two variables: `$agentip` and `$agentport`. The `/(slave|agent)/` endpoint can do `proxy_pass http://$agentip:$agentport` while you will just `proxy_pass http://$agentip:61001/system/logs/v1/$url$is_args$query_string`. We would have to double check with core people if we can rely on `agent["pid"]` always following `<type>@<ip>:<port>` format.